behaviour/graql/language/delete.feature

```diff
@@ -604,52 +978,327 @@ Feature: Graql Delete Query
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has postcode;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has postcode "W1U8ED";
+      """
+    Given the integrity is validated
+    Then graql delete throws
+      """
+      match
+        $x isa person, has address $a;
+      delete
+        $x has thing $a;
+      """
+
+
+  Scenario: you can specify the direct type of an attribute when deleting an ownership of it
+    Given get answers of graql insert
+      """
+      insert
+      $x isa person, has name "Watson";
+      """
+    Given the integrity is validated
+    Given concept identifiers are
+      |      | check | value      |
+      | WAT | key   | name:Watson |
+      | nWA | value | name:Watson |
+    Then uniquely identify answer concepts
+      | x   |
+      | WAT |
+    When graql delete
+      """
+      match
+        $x isa person;
+      delete
+        $x isa! person;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match $x isa person; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: deleting an attribute ownership throws an error when the incorrect direct type is specified
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Watson";
+      """
+    Given the integrity is validated
+    Then graql delete throws
+      """
+      match
+        $x isa person;
+      delete
+        $x isa! entity;
+      """
+
+
+  Scenario: deleting the owner of an attribute also deletes the attribute ownership
+    Given graql define
+      """
+      define
+      duration sub attribute, value long;
+      friendship has duration;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Tom";
+      $y isa person, has name "Jerry";
+      $r (friend: $x, friend: $y) isa friendship, has ref 0, has duration 1000;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has duration $d; get;
+      """
+    When concept identifiers are
+      |      | check | value         |
+      | REF0 | key   | ref:0         |
+      | DURA | value | duration:1000 |
+    Then uniquely identify answer concepts
+      | x    | d    |
+      | REF0 | DURA |
+    When graql delete
+      """
+      match
+        $r isa friendship;
+      delete
+        $r isa relation;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match $x has duration $d; get;
+      """
+    Then answer size is: 0
 
 
-  Scenario: using unmatched variable in delete throws even without data
+  Scenario: deleting the last roleplayer in a relation deletes both the relation and its attribute ownerships
```

Comment:
should this be grouped with `when the last role player is disassociated from a relation instance, the relation instance gets deleted`?

Reply:
Well, the other one tests that the relation gets cleaned up; this one primarily tests that the attribute ownerships get cleaned up. I don't think grouping them is appropriate here.